### PR TITLE
[Merged by Bors] - feat(LinearAlgebra/Matrix/Trace): add Matrix.trace_diagonal

### PR DESCRIPTION
**Pull request by adomani**
The trace of a diagonal matrix is the sum of the diagonal elements.

[Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Trace.20of.20diagonal)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ADedecker** [comment]
Is this ready for review?

**Comment by adomani** [comment]
Ah, yes, I simply forgot to tag it!

File: Mathlib/LinearAlgebra/Matrix/Trace.lean [hunk 1]
```diff
@@ -42,6 +42,10 @@ def trace (A : Matrix n n R) : R :=
   ∑ i, diag A i
 #align matrix.trace Matrix.trace
 
+lemma trace_diagonal {o} [Fintype o] [DecidableEq o] (d : o → R) :
+    trace (diagonal d) = (Finset.univ : Finset o).sum d := by
```
Comment (ADedecker) [inline Mathlib/LinearAlgebra/Matrix/Trace.lean:46]:
```suggestion
    trace (diagonal d) = ∑ i, d i := by
```
I *think* this is better, but if you have a use case where that is annoying feel free to ignore.

**Review by ADedecker — COMMENTED** [review]
bors d+

File: Mathlib/LinearAlgebra/Matrix/Trace.lean (hunk 1 above)
Comment (adomani) [reply Mathlib/LinearAlgebra/Matrix/Trace.lean:46]: I am happy to go with this spelling, thanks!

**Pull request closed without merging**